### PR TITLE
Review pass

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

File: projects/jsoup/Dockerfile [hunk 1]
```diff
@@ -15,17 +15,24 @@
 ################################################################################
 
 FROM gcr.io/oss-fuzz-base/base-builder
-RUN apt-get update && apt-get install -y maven
 
-RUN git clone --depth 1 https://github.com/google/fuzzing
-RUN cat fuzzing/dictionaries/html.dict > $SRC/JsoupFuzzer.dict
-RUN cat fuzzing/dictionaries/xml.dict > $SRC/JsoupXMLFuzzer.dict
+RUN curl -L https://downloads.apache.org/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.zip -o maven.zip && \
```
Comment (fmeum) [inline projects/jsoup/Dockerfile:19]: This installs much faster, is not affected by OS image upgrades and installs less unused dependencies.

File: projects/jsoup/Dockerfile [hunk 2]
```diff
@@ -15,17 +15,24 @@
...
+    unzip maven.zip -d $SRC/maven && \
+    rm -rf maven.zip
 
-ENV JSOUP_BRANCH=1.13.1
+ENV MVN $SRC/maven/apache-maven-3.6.3/bin/mvn
+
+RUN git clone --depth 1 https://github.com/google/fuzzing && \
```
Comment (fmeum) [inline projects/jsoup/Dockerfile:25]: Try to keep related statement grouped into one `RUN` directive and clean up unneeded resources to keep the images small.

File: projects/jsoup/Dockerfile [hunk 3]
```diff
@@ -15,17 +15,24 @@
...
+    zip -j $SRC/XmlFuzzer_seed_corpus.zip go-fuzz-corpus/xml/corpus/* && \
+    rm -rf go-fuzz-corpus
 
 RUN git clone --depth 1 https://github.com/jhy/jsoup/
 
 COPY build.sh $SRC/
-COPY JsoupFuzzer.java JsoupXMLFuzzer.java $SRC/
-WORKDIR $SRC/
-
+COPY HtmlFuzzer.java XmlFuzzer.java $SRC/
```
Comment (fmeum) [inline projects/jsoup/Dockerfile:37]: We don't have to repeat the project name in the fuzzer names and should adhere to CamelCase.

File: projects/jsoup/HtmlFuzzer.java [hunk 4]
```diff
@@ -15,12 +15,11 @@
...
 
 import com.code_intelligence.jazzer.api.FuzzedDataProvider;
-import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
 
+import org.jsoup.Jsoup;
 
-public class JsoupFuzzer {
-	public static void fuzzerTestOneInput(FuzzedDataProvider data) {
-		Document doc = Jsoup.parse(data.consumeRemainingAsString());
```
Comment (fmeum) [inline projects/jsoup/HtmlFuzzer.java:24]: We should strive to keep the fuzz target as simple as possible. I don't think we need to store the result in a local variable, or do we?

File: projects/jsoup/XmlFuzzer.java [hunk 5]
```diff
@@ -15,14 +15,12 @@
...
+
 import org.jsoup.Jsoup;
-import org.jsoup.nodes.Document;
 import org.jsoup.parser.Parser;
 
-
-public class JsoupXMLFuzzer {
-	public static void fuzzerTestOneInput(FuzzedDataProvider data) {
-		Document doc = Jsoup.parse(data.consumeRemainingAsString(), "", Parser.xmlParser());
-		doc.toString();
```
Comment (fmeum) [inline projects/jsoup/XmlFuzzer.java:26]: Is this needed to trigger bugs? If not, it's better to remove it.

File: projects/jsoup/build.sh [hunk 6]
```diff
@@ -15,18 +15,16 @@
 #
 ################################################################################
 
-MAVEN_ARGS="-P!java14+ -Dmaven.test.skip=true -Djavac.src.version=15 -Djavac.target.version=15"
+# Move seed corpus and dictionary.
+mv $SRC/{*.zip,*.dict} $OUT
```
Comment (fmeum) [inline projects/jsoup/build.sh:19]: The dictionaries and the seed corpus need to be moved to out.

**Review by 0roman — APPROVED** [review]

**Pull request merged**